src/metatrain/utils/additive/composition.py

```diff
@@ -64,13 +69,16 @@ def __init__(self, model_hypers: Dict, dataset_info: DatasetInfo):
     def train_model(
         self,
         datasets: List[Union[Dataset, torch.utils.data.Subset]],
+        additive_models: List[torch.nn.Module],
         fixed_weights: Optional[Dict[str, Dict[int, str]]] = None,
     ) -> None:
         """Train/fit the composition weights for the datasets.
 
         :param datasets: Dataset(s) to calculate the composition weights for.
         :param fixed_weights: Optional fixed weights to use for the composition model,
             for one or more target quantities.
+        :param additive_models: Additive models to be removed from the targets
```

Comment:
Why this removal should happen implicitly in the composition model class? Maybe it is better to either have a separate utility function that removes additives?